### PR TITLE
network: dhcp6pd: cleanups for unreachable route

### PR DESCRIPTION
**Pull request by yuwata**
Split out of #21625.

### CODE REVIEW

File: src/network/networkd-dhcp-prefix-delegation.c [hunk 1]
```diff
@@ -667,17 +667,17 @@ static int dhcp6_route_handler(sd_netlink *rtnl, sd_netlink_message *m, Link *li
...
         assert(addr);
 
-        (void) in6_addr_prefix_to_string(addr, prefixlen, &buf);
+        if (prefixlen >= 64) {
+                _cleanup_free_ char *buf = NULL;
 
-        if (prefixlen == 64) {
-                log_link_debug(link, "Not adding a blocking route for DHCPv6 delegated subnet %s since distributed prefix is 64",
+                (void) in6_addr_prefix_to_string(addr, prefixlen, &buf);
+                log_link_debug(link, "Not adding a blocking route for DHCPv6 delegated prefix %s since prefix length is larger or equals to 64.",
```
Comment (keszybz) [inline src/network/networkd-dhcp-prefix-delegation.c]: Grammar here. Maybe just say "since prefix has length >= 64" for brevity?

**Review by keszybz — COMMENTED** [review]
Looks reasonable.

**Comment by yuwata** [comment]
@keszybz Thanks. The log message is fixed. Upgrading the green label.

**Comment by yuwata** [comment]
The CentOS CI (Arch Linux + sanitizers) failure is caused by oomd. Unrelated.

**Pull request merged**